src/main.js

```diff
@@ -3,22 +3,21 @@ const { app, ipcMain: ipc } = electron
 
 let windows = require('./windows/windows')
 let TimerState = require('./state/timer-state')
-let writeState = require('./state/write-state')
-let readState = require('./state/read-state')
+let statePersister = require('./state/state-persister')
 
 let timerState = new TimerState()
 
 app.on('ready', () => {
   timerState.setCallback(onTimerEvent)
-  timerState.loadState(readState.read())
+  timerState.loadState(statePersister.read())

```

Comment:
I can see some reasons to do as you suggest, but I think I disagree with your arguments for why (and overall, I prefer the current loosely coupled version).

Why does the timer-state need to control its own persistence? Currently as it stands, the timer-state is just a pure model of data. Moving the control of saving/loading of that state into the timer-state doesn't remove the coupling, it just moves it. And I'd argue it makes the coupling tighter.

In the current implementation, is it really temporal coupling? The timer-state works just fine if you don't call `loadState` (you just don't get the prior state, but the app still works). If we added error handling inside the statePersister or changed things about the app startup, the timer-state would be unaffected. Similarly, if we made the file load asynchronous, everything would still be fine from a temporal-coupling standpoint -- the load would eventually broadcast the change events. (Though from a UX standpoint it's less desirable to have the thing be blinky on startup.)

As things stand right now, the read/write to disk is *not* a side effect. It isn't something that happens unexpectedly when you called a method. Rather it is an event-driven behavior. When the timer-state broadcasts an event saying it has been updated, the main.js orchestrates the persistence. (We could make it further event-driven with more of a pub/sub model, but that's a different thing too...)

What I like about the current model is that we've kept these two responsibilities decoupled. Currently we persist data every time the state changes. But it doesn't need to be that way. For example, we might want to debounce saving state because it tends to happen in bursts. But if the app closes, save anything that's unsaved w/o debouncing. That kind of change is easy to make in the current setup, but would be harder to implement if we move the persistence control into timer-state.

So I'd prefer to keep these things separate for now. I can understand the desire to push the read/write calls into timer-state, but I don't think we've found the correct arguments for it.

Reply:
My two cents: I prefer that `TimerState` not be concerned with holding state _and_ reading / writing from disk. But, I know there are good arguments on either side.

This PR has some good fixes in it that would be nice to merge - what if we merge it and then continue to discuss the question of whether or not `TimerState` should own the persistence?

Reply:
I agree. I think this is a good change and shouldn't be held up any longer.

Reply:
Agreed. I'll merge it.